modules/standard/Map.chpl

```diff
@@ -398,18 +398,17 @@ module Map {
     }
 
     /*
-      Get the value mapped to the given key, or add the mapping if key does not
-      exist.
+      Get the value mapped to the given key if the key exists in the map. If the key does not exist and the value type is default initializable, the mapping is added. Otherwise, throws an error.
```

Comment:
Let's also try to get this back to 80 character lines, please

Reply:
The last section of `doc/rst/language/spec/procedures.rst` describes the cases for the `ref` and `const ref` procs to be chosen, but I don't think there's a name for the two types of contexts that you can just refer to here.  Maybe there should be one, and then you could use it.  (Or maybe I just didn't look hard enough.)

I nominate `mutable context` and `immutable context`. as in 

```
If the key does not exist and the value type is default initializable and 
this proc is called in mutable context, the mapping is added. Otherwise, 
throws an error.
```

Reply:
Alright, here's what I've come up with based on all of the feedback:
```
      If the key exists in the map, get the value mapped to the given key.
      If the key does not exist in the map, the value type is default
      initializable, and this proc is called in a mutable context, then
      the mapping is added. If the key does not exist in the map and either
      the value type is not default initializable or this proc is called
      in a non-mutable context, then an error will be thrown.
```

So, we've got the notion of the mutable context in and clarified the "otherwise" to make it more explicit when an error will be thrown.

How does that sound?

Reply:
I'll reiterate that I just made up the phrase "in a mutable context" this morning, and that it's not defined anywhere.  I didn't mean to suggest it was suitable to go into a doc today.